MobileWeather/app/src/main/java/com/sdl/mobileweather/activity/MainActivity.java

```diff
@@ -137,21 +141,25 @@ else if ((getResources().getString(R.string.drawer_item_about)).equals(item)){
     	mDrawerList.setItemChecked(position, false);
         mDrawerLayout.closeDrawer(mDrawerList);
     }
-	
-    
-    
+
 	private void checkForCrashes() {}
 
 	private void checkForUpdates() {}
 
-	private boolean checkPermissions() {
+	private boolean checkPermission() {
 		boolean permissionsGranted;
 
 		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
 			boolean bluetoothGranted = PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.BLUETOOTH_CONNECT);
 			boolean locationGranted = PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION);
 			permissionsGranted =  (BuildConfig.TRANSPORT.equals("TCP") || bluetoothGranted) && locationGranted;
 		}
+		else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+			boolean bluetoothGranted = PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.BLUETOOTH_CONNECT);
+			boolean locationGranted = PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION);
+			boolean postNotificationsGranted = PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.POST_NOTIFICATIONS);
+			permissionsGranted =  (BuildConfig.TRANSPORT.equals("TCP") || bluetoothGranted) && locationGranted && postNotificationsGranted;
+		}
```

Comment:
The Post notification permission does not prevent that app form from working, just notifications not appearing, Going forward, we are only going to use this method to check if we can start SDL after we requested permissions. We could also rename this method hasPermisssions() or something else as we are adding a different checkPermissions() method below.
```suggestion
```